yt/chyt/controller/internal/api/api.go

```diff
@@ -780,3 +803,73 @@ func (a *API) DescribeOptions(ctx context.Context, alias string) ([]strawberry.O
 
 	return append(strawberryOptions, ctlOptions...), nil
 }
+
+func (a *API) GetSecrets(ctx context.Context, alias string) (secrets map[string]any, err error) {
+	if err := a.CheckExistence(ctx, alias, true /*shouldExist*/); err != nil {
+		return nil, err
+	}
+	if err := a.CheckPermissionToOp(ctx, alias, yt.PermissionManage); err != nil {
+		return nil, err
+	}
+
+	secretsPath := a.cfg.AgentInfo.StrawberryRoot.JoinChild(alias, "secrets")
+	secretsNodeExists, err := a.Ytc.NodeExists(ctx, secretsPath, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if secretsNodeExists {
+		err = a.Ytc.GetNode(
+			ctx,
+			secretsPath,
+			&secrets,
+			nil)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		secrets = make(map[string]any)
+	}
+
+	return
+}
+
+func (a *API) SetSecrets(ctx context.Context, alias string, secrets map[string]any) error {
+	if err := a.CheckExistence(ctx, alias, true /*shouldExist*/); err != nil {
+		return err
+	}
+	if err := a.CheckPermissionToOp(ctx, alias, yt.PermissionManage); err != nil {
+		return err
+	}
+
+	secretsPath := a.cfg.AgentInfo.StrawberryRoot.JoinChild(alias, "secrets")
+	secretsNodeExists, err := a.Ytc.NodeExists(ctx, secretsPath, nil)
+	if err != nil {
+		return err
+	}
+	if !secretsNodeExists {
```

Comment:
A few notes:
1) we can remove the `NodeExists` check if we add `force` flag to the `CreateNode`
2) we create the `secrets` node in two places and it has non-trivial copy-pasted attributes (acl/inherit_acl). We can move it to some helper and call it from both places (e.g. `func (a *API) createSecretsNode(...)`).